### DISCUSSION TITLE
Add missing 'faces' to database ?

### DISCUSSION THREAD

**Discussion by ceejayemm** [photoprism/photoprism#2505] [Q&A]
Is there a way to manually add people to the Faces database ?  I have indexed my images and the automatic face recognition has done a pretty good job of identifying faces in my images. It does vary though, sometimes it finds blurry faces in mirror/reflections, sometime right at the back of a crowd. On other occasions it misses what to me seem pretty clear faces. For example I have an image of a group of three friends. Two of these friends have been identified and added to the faces database. The third friend has not, despite already being in the faces database, and I don't really know why. 

I am using PhotoPrism with its default, out of the box, settings at the moment and so haven't tweaked anything. Is there anything I can do to improve the automatic face recognition in this case (without ending up with more 'odd' faces from crowds, mirrors etc). Failing that, can I manually add a reference to the third friend in this image to their existing face database entry ?

I am currently using PP Build 220629-5d7448d2 via Docker on a Debian 11 host with MariaDB 10.8 as the backend database. I can send you the relevant image if that would help but would prefer not to put it on an public forum post - for obvious reasons.

Thanks

Chris

**Comment by graciousgrey** [comment]
You mean the face was not detected at all ?

Possible reasons for this are listed here:

https://docs.photoprism.app/user-guide/organize/people/#when-a-face-was-not-detected

Manual tagging of faces is not yet implemented, but planned: 
https://github.com/photoprism/photoprism/issues/1548

**Comment by Yabbo** [comment]
Whats odd is i have an entire album that is mostly portraits and they are even labeled automatically as such but for some reason they dont say there is a face in the picture.
